tv_apps/browser/js/bookmarkList.js

```diff
@@ -0,0 +1,839 @@
+
+      itemEl.classList.add('list-item');
+      itemEl.setAttribute('tabindex', '0');
+      itemEl.setAttribute('data-type', data.type);
+      if(data.readOnly) {
+        itemEl.setAttribute('readOnly', data.readOnly);
+      } else {
+        itemEl.setAttribute('readOnly', false);
+      }
+
+      iconLeftEl.classList.add('icon-box', 'icon-box-left');
+      iconImgLeftEl.classList.add('icon');
+      iconLeftEl.appendChild(iconImgLeftEl);
+
+      textEl.classList.add('text-box');
+      titleEl.classList.add('title');
+      titleEl.textContent = data.title;
+      urlEl.classList.add('uri');
+      textEl.appendChild(titleEl);
+      textEl.appendChild(urlEl);
+
+      iconRightEl.classList.add('icon-box', 'icon-box-right');
+      iconImgRightEl.classList.add('icon');
+      iconRightEl.appendChild(iconImgRightEl);
+
+      if(data.type) {
+        itemEl.setAttribute('data-type', data.type);
+        switch (data.type) {
+          case 'folder':
+            itemEl.setAttribute('data-folder', data.id);
+            iconImgLeftEl.setAttribute('data-icon', 'folder');
+            iconImgRightEl.setAttribute('data-icon', 'arrow-right');
+            break;
+          case 'bookmark':
+            itemEl.setAttribute('data-folder', '');
+            if(data.iconUri) {
+              iconImgLeftEl.setAttribute('data-icon', '');
+              iconImgLeftEl.style.backgroundImage = 'url(' + data.iconUri +')';
+            } else {
+              iconImgLeftEl.setAttribute('data-icon', 'default-fav');
+            }
+            urlEl.textContent = data.uri;
+            iconImgRightEl.setAttribute('data-icon', '');
+            break;
+          case 'button':
+            itemEl.setAttribute('data-folder', '');
+            iconImgLeftEl.setAttribute('data-icon', 'folder');
+            iconImgRightEl.setAttribute('data-icon', 'arrow-left');
+            break;
+          default:
+            break;
+        }
+      } else {
+        itemEl.setAttribute('data-type', 'bookmark');
+        itemEl.setAttribute('data-folder', '');
+        if(data.iconUri) {
+          iconImgLeftEl.setAttribute('data-icon', '');
+          iconImgLeftEl.style.backgroundImage = 'url(' + data.iconUri +')';
+        } else {
+          iconImgLeftEl.setAttribute('data-icon', 'default-fav');
+        }
+        urlEl.textContent = data.uri;
+        iconImgRightEl.setAttribute('data-icon', '');
+      }
+
+
+      itemEl.appendChild(iconLeftEl);
+      itemEl.appendChild(textEl);
+      itemEl.appendChild(iconRightEl);
+
+      itemEl.
+        addEventListener('keydown', this.handleItemKeyDown.bind(this));
+      itemEl.
+        addEventListener('keyup', this.handleItemKeyUp.bind(this));
+      itemEl.
+        addEventListener('mouseup', this.handleItemMouseUp.bind(this));
+      itemEl.
+        addEventListener('mouseover', this.handleItemMouseOver.bind(this));
+      itemEl.
+        addEventListener('mouseout', this.handleItemMouseOut.bind(this));
```

Comment:
Adding too many event listeners will consume memory. Please add ```// XXX: delegate these events later``` and file a follow up bug for it.